Dynatrace/dynatrace-operator
# Mount custom eec tls certificate

## Description

- This PR implements Mount custom EEC TLS certificate to eec sts and otel sts.

## How can this be tested?

1. Create a secret with custom TLS for eec

```
openssl req -nodes -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365 -subj "/CN=dynakube-extensions-controller.dynatrace"
kubectl create secret tls custom-tls-cert --cert=cert.pem --key=key.pem
```

2. Deploy dynakube:

```
spec:
  extensions:
    prometheus:
      enabled: true
  templates:
    extensionExecutionController:
      tlsRefName: custom-tls-cert
      imageRef:
        repository: extk8sregistry.azurecr.io/eec/dynatrace-eec
        tag: 1.300.0.20240814-073129
```

3. Make sure all secret are mounted properly and env vars are there based on 
ticket description [K8S-10666](https://dt-rnd.atlassian.net/browse/K8S-10666)
